internal/Connect-SqlServer.ps1

```diff
@@ -10,7 +10,8 @@ Internal function that creates SMO server object. Input can be text or SMO.Serve
 		[object]$SqlServer,
 		[System.Management.Automation.PSCredential]$SqlCredential,
 		[switch]$ParameterConnection,
-		[switch]$RegularUser
+		[switch]$RegularUser,
+		[string]$ApplicationName
```

Comment:
Hey Stuart - this looks great and will be approved as soon as one thing is changed!

Do this `[string]$ApplicationName = "dbatools PowerShell module - dbatools.io"` and you wont have to do any checks and you can always just set `ConnectionContext.ApplicationName = $ApplicationName`